app/services/experiment_control_service.rb

```diff
@@ -0,0 +1,94 @@
+        .where("appointments.status = ?", "scheduled")
+        .where("appointments.scheduled_date BETWEEN ? AND ?", experiment_start, experiment_end)
+        .order(Arel.sql("random()"))
+
+      experiment_patients.each do |patient|
+        group = experiment.group_for(patient.id)
+        appointment = patient.appointments
+          .where(status: "scheduled")
+          .where("appointments.scheduled_date BETWEEN ? AND ?", experiment_start, experiment_end).first
+        schedule_reminders(patient, appointment, experiment, group, appointment.scheduled_date)
+        Experimentation::TreatmentGroupMembership.create!(treatment_group: group, patient: patient)
+      end
+
+      experiment.update!(state: "live", start_date: experiment_start, end_date: experiment_end)
+    end
+
+    def start_stale_patient_experiment(name, total_days)
+      experiment = Experimentation::Experiment.find_by!(name: name, experiment_type: "stale_patient_reminder")
+      date = Date.current
+      eligibility_start = (date - 365.days).beginning_of_day
+      eligibility_end = (date - 35.days).end_of_day
+
+      experiment.state_selecting!
+
+      eligible_patients = patient_pool
+        .joins(:encounters)
+        .where("encounters.device_created_at BETWEEN ? AND ?", eligibility_start, eligibility_end)
+        .where("NOT EXISTS (SELECT 1 FROM encounters WHERE encounters.patient_id = patients.id AND
+                encounters.device_created_at > ?)", eligibility_end)
+        .left_joins(:appointments)
+        .where("NOT EXISTS (SELECT 1 FROM appointments WHERE appointments.patient_id = patients.id AND
+                appointments.scheduled_date >= ?)", date)
+        .order(Arel.sql("random()"))
+        .to_a
+
+      total_days.times do
+        daily_patients = eligible_patients.pop(10_000)
+        break if daily_patients.empty?
+        daily_patients.each do |patient|
+          group = experiment.group_for(patient.id)
+          appointment = patient.appointments.where(status: "scheduled").last
+          schedule_reminders(patient, appointment, experiment, group, date)
+          Experimentation::TreatmentGroupMembership.create!(treatment_group: group, patient: patient)
+        end
+        date += 1.day
+      end
+
+      experiment.update!(state: "live", start_date: Date.current, end_date: total_days.days.from_now.to_date)
+    end
+
+    protected
+
+    def patient_pool
+      Patient.from(Patient.with_hypertension, :patients)
+        .contactable
+        .where("age >= ?", 18)
+        .includes(treatment_group_memberships: [treatment_group: [:experiment]])
+        .where(["experiments.end_date < ? OR experiments.id IS NULL", 14.days.ago]).references(:experiment)
+    end
+
+    def schedule_reminders(patient, appointment, experiment, group, schedule_date)
+      group.reminder_templates.each do |template|
+        remind_on = schedule_date + template.appointment_offset.days
+        AppointmentReminder.create!(
```

Comment:
This could hang off the experiment to make things more clear:

```
experiment.appointment_reminders.create! ....
```

Reply:
Experiment does not currently have a relationship with appointment reminders. I added the experiment id to appointment reminder at the last minute as part of the other PR but forgot to add the `has_many` to the experiment. But you suggested getting rid of the explicit relationship altogether here:
`https://github.com/simpledotorg/simple-server/pull/2177/files#r603542076`